### DISCUSSION TITLE
How to update CAPH / Cluster API?

### DISCUSSION THREAD

**Discussion by privatecoder** [syself/cluster-api-provider-hetzner#643] [Q&A] [answered]
Hi,

just wanted to ask: Is it safe to upgrade the caph-controller-manager from a previous Beta, for example Beta 9, to a more recent version?

Also: What is the procedure to upgrade Cluster API to a supported version or is it upgraded along with the controller-manager?

Thank you!

**Comment by batistein** [comment, marked as answer]
I just did recently an update of beta.10 to the current beta.14 without issues.

I used the following way:
1. Update capi-stack via `clusterctl upgrade plan` followed by `clusterctl upgrade apply --contract v1beta1`
2. Update the caph controller via: ` clusterctl upgrade apply --infrastructure cluster-api-provider-hetzner-system/hetzner:v1.0.0-beta.14`

In general please check out the release note for changes. And of course if you want to be safe I would recommend a version by version update. These are always tested via e2e tests.

**Reply by privatecoder** [reply]
This worked like a charm, upgrading capi-stack first, followed by upgrading caph firt to .10 and then to .14. Thank you!

PS: Should we add this to the docs?

**Reply by batistein** [reply]
Yes feel free to open a PR ;)

**Reply by guettli** [reply]
The docs now have a text about upgrading CAPH: https://github.com/syself/cluster-api-provider-hetzner/blob/main/docs/topics/upgrade.md
